### ISSUE TITLE
PrecacheController should await potentially async plugin cacheWillUpdate

### ISSUE THREAD

**Issue by prichey** [GoogleChrome/workbox#2285]
According to [the docs](https://developers.google.com/web/tools/workbox/guides/using-plugins#custom_plugins) a plugin's `cacheWillUpdate` can be asynchronous but this function is not `await`ed in the PrecacheController ([src](https://github.com/GoogleChrome/workbox/blob/8b22c152c6c30e6b50097bf76503fdd8c38e8378/packages/workbox-precaching/src/PrecacheController.ts#L265)). I believe this is leading to non-deterministic behavior in the installation process.

Here's a reconstruction of my logs, if it's helpful:
```
Installing the PrecacheController...
Network request for '/foo.js' returned a response with status '200'.
customPlugin.cacheWillUpdate called for '/foo.js'.
customPlugin.cacheWillUpdate called for '/foo.js'.
Updating the 'mock' cache with a new Response for /foo.js.
Successfully installed the PrecacheController
```

**Comment by jeffposnick** [comment]
Thanks—you're correct, and we'll get that sorted.

**Comment by prichey** [comment]
Thanks for addressing this so quickly!

Is the plan to merge minor changes into the 5.0 release or do you anticipate releasing a 4.3.2?

**Comment by jeffposnick** [comment]
This will be released in the near future as part of `v5.0.0-rc.1`.

If you're still on `v4.3.1`, you can work around the issue by implementing your own, non-`async` `cacheWillUpdate` plugin:

```
const customPlugin = {
  cacheWillUpdate: ({request, response, event}) => {
    // Examine response and either return it, or null.
  },
}

PrecacheController.install({plugins: [customPlugin]});
```

(That's assuming that you're able to make a synchronous decision about whether to cache the response, i.e. you don't need to read its body or something else that's inherently promise-based.)

**Comment by prichey** [comment]
Unfortunately I need to look at the body of the response so I'll need to keep `cacheWillUpdate` asynchronous.
